optuna/samplers/_grid.py

```diff
@@ -185,6 +179,23 @@ def sample_independent(
 
         return param_value
 
+    def after_trial(
+        self,
+        study: Study,
+        trial: FrozenTrial,
+        state: TrialState,
+        values: Optional[Sequence[float]],
+    ) -> None:
+        target_grids = self._get_unvisited_grid_ids(study)
+
+        if len(target_grids) == 0:
+            study.stop()
+        elif len(target_grids) == 1:
+            if state.is_finished():
+                grid_id = study._storage.get_trial_system_attrs(trial._trial_id)["grid_id"]
+                if grid_id == target_grids[0]:
+                    study.stop()
```

Comment:
```suggestion
            grid_id = study._storage.get_trial_system_attrs(trial._trial_id)["grid_id"]
            if grid_id == target_grids[0]:
                study.stop()
```

Thank your for applying my previous suggestion.

I've noticed that `if state.is_finished():` is now redundant because the state should be either of `COMPLETE`, `PRUNED`, or `FAIL` as long as `after_trial()` is triggered.

Sorry for suggesting for the same point twice.

Reply:
You're right. Thanks.